File: 백준 5639 TR 이진 검색 트리.py
import sys

sys.setrecursionlimit(100000)

input = sys.stdin.readline


def post_order(start, end):
    if start > end:
        return

    div = end + 1
    for i in range(start + 1, end + 1):
        if data[start] < data[i]:
            div = i
            break

    post_order(start + 1, div - 1)
    post_order(div, end)
    print(data[start])


data = []
while True:
    try:
        data.append(int(input().strip()))
    except:
        break

post_order(0, len(data) - 1)

File: 백준 5639 TR 이진 검색 트리.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let data = [];

rl.on('line', (line) => {
    data.push(parseInt(line));
}).on('close', () => {
    postOrder(0, data.length - 1);
    process.exit();
});

function postOrder(start, end) {
    if (start > end) return;

    let div = end + 1;
    for (let i = start + 1; i <= end; i++) {
        if (data[start] < data[i]) {
            div = i;
            break;
        }
    }

    postOrder(start + 1, div - 1);
    postOrder(div, end);
    console.log(data[start]);
}
